terps/scare/os_glk.c

```diff
@@ -1737,6 +1737,85 @@ os_stop_sound (void)
 }
 
 
+#ifdef GARGLK
+static const char *gamefile;
+
+struct GlkImage {
+  sc_int offset;
+  int id;
+};
+
+static struct GlkImage glk_images[8192];
+static size_t glk_images_count = 0;
+
+/*
+ * os_show_graphic()
+ *
+ * Use the Gargoyle-specific garglk_add_image_resource() to load images from memory.
+ */
+void
+os_show_graphic (const sc_char *filepath, sc_int offset, sc_int length)
+{
+  FILE *in = NULL;
+  unsigned char *data = NULL;
+
+  for (size_t i = 0; i < glk_images_count; i++) {
+    if (glk_images[i].offset == offset) {
```

Comment:
What happens if multiple files have an image at the same offset? Or is Scare not one of the terps that can have multiple resource files?

Reply:
That's an excellent point that I never considered.

Based on the earlier code, it looks like this is always looked up in the main game file. On line 3560:

```
#ifdef LINUX_GRAPHICS
  /* Note the path to the game file for graphics extraction. */
  gsclinux_game_file = argv[argv_index];
#endif
```
So this is safe.

But SCARE _does_ seem to be able to use different resource files!  So you're absolutely right, the equivalent code in the SCARE interpreter is not correct.  I'll get that fixed up.  Nice find!

Reply:
This comment was on Scare 😉

Reply:
Sorry, I mean _Hugo_ can have different resource files, not SCARE!